server/view.go

```diff
@@ -95,6 +101,138 @@ func serveView(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func serveBatView(w http.ResponseWriter, r *http.Request) {
+	err := r.ParseForm()
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(`{"error":"invalid URL query"}`))
+		return
+	}
+
+	GameID, err := strconv.Atoi(r.Form.Get("id"))
+
+	if err != nil || GameID <= 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(`{"error":"invalid id value"}`))
+		return
+	}
+
+	recordingPosition := getRecordingPositionByGameId(strconv.Itoa(GameID))
```

Comment:
Please use the `recordings` global map instead. See: https://github.com/1lann/lol-replay/blob/c7ec366fa41f8b4e70af8d5febcfb2d87cff62af/server/server.go#L233

Please acquire the `recordingsMutex` as well when using it.